metrics/src/datapoint.rs

```diff
@@ -1,9 +1,51 @@
+//! This file defines a set of macros for reporting metrics.
+//!
+//! To report a metric, simply calling one of the following datapoint macros
+//! with a suitable message level:
+//!
+//! - datapoint_debug!
+//! - datapoint_info!
+//! - datapoint_trace!
+//! - datapoint_warn!
+//! - datapoint_error!
```

Comment:
nit: What do you think of reorder by severity (I think this is the order they appear in the code) ?
- error
- warn
- info
- debug
- trace